lib/proxy.js

```diff
@@ -85,10 +85,16 @@ Proxy.prototype.listen = function(options, callback) {
         self.httpsServer = httpsServer;
         self.wssServer = wssServer;
         self.httpsPort = port;
-        self.httpServer.listen(listenOptions, callback);
+        self.httpServer.listen(listenOptions, () => {
+          self.httpPort = self.httpServer.address().port;
+          callback();
+	});
       });
     } else {
-      self.httpServer.listen(listenOptions, callback);
+      self.httpServer.listen(listenOptions, () => {
+        self.httpPort = self.httpServer.address().port;
```

Comment:
Won't this silently swallow error if `listen()` fails?

Reply:
If `listen()` fails, the callback won't be called at all. So if the callback is being called, it means `listen()` has succeeded. In order to catch the error, we should listen on the `error` event.
```js
self.httpServer.on('error', e => ...)
```

https://nodejs.org/api/net.html#net_server_listen

Reply:
Ah I see - I think that'd be a good improvement.